keras_nlp/samplers/top_p_sampler.py

```diff
@@ -65,17 +67,28 @@ def next(prompt, cache, index):
     def __init__(
         self,
         p=0.1,
+        sample_from_top_k=40,
```

Comment:
I think this does need to default to `None` for now. We could try to set a default value here, but I think it would need to depend on both `p` and the vocabulary size, and be a somewhat interesting research question to set right. Let's default to None and leave this as end user config for now.